scripts/mysql_system_tables.sql

```diff
@@ -104,7 +104,7 @@ set @had_user_table= @@warning_count != 0;
 
 CREATE TABLE IF NOT EXISTS roles_mapping ( Host char(60) binary DEFAULT '' NOT NULL, User char(80) binary DEFAULT '' NOT NULL, Role char(80) binary DEFAULT '' NOT NULL, Admin_option enum('N','Y') COLLATE utf8_general_ci DEFAULT 'N' NOT NULL, UNIQUE (Host, User, Role)) engine=Aria transactional=1 CHARACTER SET utf8 COLLATE utf8_bin comment='Granted roles';
 
-CREATE TABLE IF NOT EXISTS func (  name char(64) binary DEFAULT '' NOT NULL, ret tinyint(1) DEFAULT '0' NOT NULL, dl char(128) DEFAULT '' NOT NULL, type enum ('function','aggregate') COLLATE utf8_general_ci NOT NULL, PRIMARY KEY (name) ) engine=Aria transactional=1 CHARACTER SET utf8 COLLATE utf8_bin   comment='User defined functions';
+CREATE TABLE IF NOT EXISTS func (  name char(64) COLLATE utf8_bin NOT NULL CHECK (name <> ''), ret tinyint(1) DEFAULT '0' NOT NULL CHECK (ret < 5 and ret >= 0 and ret <> 3), dl char(128) COLLATE utf8_bin NOT NULL CHECK (dl <> ''), type enum ('function','aggregate') COLLATE utf8_general_ci NOT NULL, PRIMARY KEY (name) ) engine=Aria transactional=1 CHARACTER SET utf8 COLLATE utf8_bin   comment='User defined functions';
```

Comment:
Why exactly to change collation ?
And what is reason for setting the check of the `ret`? Also which logic you used for this set and why?

Reply:
It was me that suggested the `ret` constraints above. The values correspond the the value types that UDF functions can return. For completeness this should be in the commit message.

Something like
"return values (ret) can be  STRING_RESULT(0), REAL_RESULT(1), INT_RESULT(2), and DECIMAL_RESULT(4), but not ROW_RESULT(3) or TIME_RESULT(5)".

The commit message should also have a title that begins with 'MDEV-23299:......'

I just checked and udf can't return TIME_RESULT(5).

What do you think about `ret <>3 AND ret BETWEEN 0 AND 4)`? read-ability is the criteria.